### ISSUE TITLE
Hands on Lab Exercise 2 - Client side has multiple errors

### ISSUE THREAD

**Issue by laultman** [MicrosoftDocs/windows-uwp#2746]
Have been working my way through the HOL with very good results. The second exercise has multiple errors that cause the project build to fail.

Missing are return type.
`public static async Task<bool> CreatePassportKeyAsync(Guid userId, string username)
{
    KeyCredentialRetrievalResult keyCreationResult = await KeyCredentialManager.RequestCreateAsync(username, KeyCredentialCreationOption.ReplaceExisting);
}`

Not initialized anywhere: _isExistingLocalAccount
`private async void SignInPassportAsync()
{
    if (_isExistingLocalAccount)
    {`

Calls the wrong method
`        private void PassportSignInButton_Click(object sender, RoutedEventArgs e)
        {
            ErrorMessage.Text = "";
            SignInPassport();
        }
`

Because of the errors the HOL becomes unusable because the flow and intent of the lab is broken.
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7bcd30a1-2df4-6eed-a744-a7e29369197f
* Version Independent ID: 40c4f8a1-8c4c-c252-048d-13fbc1b6b49a
* Content: [Create a Windows Hello login service - UWP applications](https://docs.microsoft.com/en-us/windows/uwp/security/microsoft-passport-login-auth-service)
* Content Source: [windows-apps-src/security/microsoft-passport-login-auth-service.md](https://github.com/MicrosoftDocs/windows-uwp/blob/docs/windows-apps-src/security/microsoft-passport-login-auth-service.md)
* Product: **uwp**
* Technology: **platform**
* GitHub Login: @hickeys
* Microsoft Alias: **hickeys**

**Comment by laultman** [comment]
The instructions say to "update" SignInPassport method but the example code has an new method SignInPassportAsync. The instructions should be changed to "remove existing SignInPassport method and replace with SignInPassportAsync method. Then change PassportSignInButton_Click method call from SignInPassport to SignInPassportAsync"

**Comment by laultman** [comment]
This language is incorrect for the code snippet provided.
"As Windows Hello will create a different public and private key pair for each account on each device the Welcome page will need to display a list of registered devices for the logged in account, and allow each one to be forgotten. In Welcome.xaml add in the following XAML underneath the ForgetButton. This will implement a forget device button, an error text area and a list to display all devices."
Given the code snippet the user should replace the entire Grid section. Adding the **following underneath the ForgetButton** is not the intended result. Need more concise language.

**Comment by laultman** [comment]
This login.xaml.cs code does not work, "inaccessible due to protection level" the UserListView is a control. Setting the x:FieldModifier="Public" does not work in this case.
`if (accounts.Any())
            {
                UserListView.ItemsSource = accounts;
                UserListView.SelectionChanged += UserSelectionChanged;
            }
            else`

**Comment by laultman** [comment]
Whoever wrote the first hands on lab section did an excellent job. Can't say that for the second HOL. The instructions are poorly written, somewhat confusing, inaccurate, and the code does not work in the examples. The HOL appears to have been just thrown together and published without testing.

**Comment by laultman** [comment]
I admit, I had a cut/paste error in Login.xaml.cs that caused the "This login.xaml.cs code does not work, "inaccessible due to protection level" the UserListView is a control. Setting the x:FieldModifier="Public" does not work in this case." That code was accidentally copied from UserSelection.xaml.cs.
The code still would not build. The local variable _isExistingAccount is incorrectly called _isExistingLocalAccount

**Comment by laultman** [comment]
To get the code to build for the first run I patched the code with a temporary fix.
        public static async Task<bool> CreatePassportKeyAsync(Guid userId, string username)
        {
            KeyCredentialRetrievalResult keyCreationResult = await KeyCredentialManager.RequestCreateAsync(username, KeyCredentialCreationOption.ReplaceExisting);
            return (keyCreationResult.Status == KeyCredentialStatus.Success);
        }

The next section after the first run addresses this issue in the HOL.

**Comment by alvinashcraft** [comment]
Hi @laultman. I apologize for the extremely long delay on this issue. I appreciate all the details you've included in your comments. I walked through both HOL pages myself today and made a list of issues I found on both pages. I'll start working on fixes soon, and I'm also planning to rename the whole thing to not use the Microsoft Passport name in the code anymore. I'll update it all to name things with Windows Hello instead. I'll report back when my changes are ready.
